### ISSUE TITLE
[Question] Is grpc-java still 1.6 Compatibility 

### ISSUE THREAD

**Issue by wu-sheng** [grpc/grpc-java#3123]
I found out in https://github.com/grpc/grpc-java/blob/master/build.gradle#L50, 
```
    sourceCompatibility = 1.6
    targetCompatibility = 1.6
```

But in GRPC quick-start website, http://www.grpc.io/docs/quickstart/java.html, it required for **version 7 or higher**

Which one is right? It is different because of runtime and compile-time?

**Comment by ericgribkoff** [comment]
gRPC is compatible with Java 6. The quick-start guides have JDK version 7 as a prerequisite because the examples use gradle, which has required Java 7+ to run since the end of 2013. gradle can still build targets for Java 6, the tool itself just needs Java 7 to run - this doesn't effect gRPC's compatibility.

**Comment by wu-sheng** [comment]
Greatful, sky-walking open source tracing system will use it instead of rest service.

**Comment by ejona86** [comment]
It's more nuanced than that, as the README says:
> gRPC-Java works with JDK 6. TLS usage typically requires using Java 8, or Play Services Dynamic Security Provider on Android. Please see the Security Readme.

JDK 6 source/target support today is a way to make it easy to know we run on Android.

@wu-sheng, there has been discussion before about bumping the requirement to JDK 7 (or really, the Android subset of Java 7). I'd be very interested to learn what users are using a Java 6 JRE today (vs just JDK 6 bytecode).

**Comment by wu-sheng** [comment]
[sky-walking](https://github.com/wu-sheng/sky-walking) is an open source distributed tracing system, like Zipkin, but based on javaagent tech, trace in auto-instrumentation mode. So I want to keep the jre requirement as low as possible. All our 14 released version, I stay in 1.6. 

Some telecom system, existed a lot of years, they are still running in jre 1.6. New system didn't have this requirement.
